### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'toString')

### DISCUSSION THREAD

**Discussion by maaft** [webpack/webpack#17101] [Q&A]
I transferred my code to a new fresh cra-repo with craco 7.1. 

After issuing `yarn start`, I get below error, which seems to appear in the `webpack` module (not in my code).

My code worked previously without any issues with `webpack 4`, `react-scripts 4` and `craco 6`.

```
hub/node_modules/react-scripts/scripts/start.js:19
  throw err;
  ^

TypeError: Cannot read properties of undefined (reading 'toString')
    at hub/node_modules/webpack/lib/cache/PackFileCacheStrategy.js:1217:53
```

Any idea how I can debug this further?

**Comment by snitin315** [comment]
Hi @maaft, thanks for creating the issue.

It seems to be coming from the following line:
https://github.com/webpack/webpack/blob/1f59cbec16c5b49c01e332d9e29373d096b69c49/lib/cache/PackFileCacheStrategy.js#L1217

For some reason, `etag` is undefined in your case. However, as per the types, it can only be `Etag` or `null`. And we have proper checks to handle when it is `null`.

Can you share a minimal repo for us to reproduce this issue?

**Comment by alexander-akait** [comment]
Please check your plugins are compatibility with webpack 5, run `npm ls webpack` and check you don't have `webpack@4` in deps

**Comment by jefflewis** [comment]
I updated `react-scripts` from 4 to 5 and saw this as well. I didn't have webpack spec'd in `package.json` and `npm ls webpack` shows it being resolved to `webpack@5.83.1`.

**Reply by hainachen** [reply]
so how do you solve it?

**Comment by livingstonex** [comment]
@jefflewis, @alexander-akait did you later get a solution to this?

**Reply by alexander-akait** [reply]
Sorry not, without code I can't debug it and provide a solution

**Comment by maaft** [comment]
Hello everyone! I found the issue. I imported an outdated version of semantic ui css. After updating (or using the official CDN link) the issue was gone.

I wonder though if webpack could have pointed me to the source of the error instead of giving me this cryptic etag error above?

Shall I attach the css file here so you can dig deeper?

**Reply by livingstonex** [reply]
@maaft Please which css file did you import wrongly? as well as the path of the file please...

**Reply by maaft** [reply]
here it is. Imported with
**src/app.css**
```css
@import "./style/font.css";
@import "./style/semantic-ui.css";
```
[style.zip](https://github.com/webpack/webpack/files/11797433/style.zip)

**Reply by alexander-akait** [reply]
@maaft It should work without, can you create reproducible test repo?

**Reply by bburns** [reply]
@maaft thanks, this fixed it for me also - 

    yarn add semantic-ui-react@latest semantic-ui-css@latest

**Reply by alexander-akait** [reply]
Very strange because it should work without updating this deps

**Reply by palashkaria** [reply]
We are facing this exact same issue when upgrading to react-scripts v5, and we currently have a copy of semantic's 1.x CSS (heavily modified), so can't do the above fix (we don't have `semantic-ui-css` as a dep, directly import CSS from our filesystem). When I remove this semantic CSS import completely, it complies. So, we can narrow down to the problem CSS file.
I will try to create a repro repo and share soon

**Reply by palashkaria** [reply]
Update: I have narrowed it down to a huge base64 URL which is being used for a font-family src by semantic-ui's CSS:

Once I remove this, it compiles:
<img width="1077" alt="image" src="https://github.com/webpack/webpack/assets/2976363/4fe6851a-11d7-47f7-86d5-cb12e9153810">
Since we are not using this component, I can safely remove it. Leaving this here in case it helps!

**Reply by DaljeetSingh1320** [reply]
Can we remove these also?

@font-face {
 font-family: Accordion;
src: url(data:application/x-font-ttf;charset=utf-8;base64,AAEAAAALAIAAAwAwT1MvMggjB5AAAAC8AAAAYGNtYXAPfOIKAAABHAAAAExnYXNwAAAAEAAAAWgAAAAIZ2x5Zryj6HgAAAFwAAAAyGhlYWT/0IhHAAACOAAAADZoaGVhApkB5wAAAnAAAAAkaG10eAJuABIAAAKUAAAAGGxvY2EAjABWAAACrAAAAA5tYXhwAAgAFgAAArwAAAAgbmFtZfC1n04AAALcAAABPHBvc3QAAwAAAAAEGAAAACAAAwIAAZAABQAAAUwBZgAAAEcBTAFmAAAA9QAZAIQAAAAAAAAAAAAAAAAAAAABEAAAAAAAAAAAAAAAAAAAAABAAADw2gHg/+D/4AHgACAAAAABAAAAAAAAAAAAAAAgAAAAAAACAAAAAwAAABQAAwABAAAAFAAEADgAAAAKAAgAAgACAAEAIPDa//3//wAAAAAAIPDZ//3//wAB/+MPKwADAAEAAAAAAAAAAAAAAAEAAf//AA8AAQAAAAAAAAAAAAIAADc5AQAAAAABAAAAAAAAAAAAAgAANzkBAAAAAAEAAAAAAAAAAAACAAA3OQEAAAAAAQASAEkAtwFuABMAADc0PwE2FzYXFh0BFAcGJwYvASY1EgaABQgHBQYGBQcIBYAG2wcGfwcBAQcECf8IBAcBAQd/BgYAAAAAAQAAAEkApQFuABMAADcRNDc2MzIfARYVFA8BBiMiJyY1AAUGBwgFgAYGgAUIBwYFWwEACAUGBoAFCAcFgAYGBQcAAAABAAAAAQAAqWYls18PPPUACwIAAAAAAM/9o+4AAAAAz/2j7gAAAAAAtwFuAAAACAACAAAAAAAAAAEAAAHg/+AAAAIAAAAAAAC3AAEAAAAAAAAAAAAAAAAAAAAGAAAAAAAAAAAAAAAAAQAAAAC3ABIAtwAAAAAAAAAKABQAHgBCAGQAAAABAAAABgAUAAEAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAADgCuAAEAAAAAAAEADAAAAAEAAAAAAAIADgBAAAEAAAAAAAMADAAiAAEAAAAAAAQADABOAAEAAAAAAAUAFgAMAAEAAAAAAAYABgAuAAEAAAAAAAoANABaAAMAAQQJAAEADAAAAAMAAQQJAAIADgBAAAMAAQQJAAMADAAiAAMAAQQJAAQADABOAAMAAQQJAAUAFgAMAAMAAQQJAAYADAA0AAMAAQQJAAoANABaAHIAYQB0AGkAbgBnAFYAZQByAHMAaQBvAG4AIAAxAC4AMAByAGEAdABpAG4AZ3JhdGluZwByAGEAdABpAG4AZwBSAGUAZwB1AGwAYQByAHIAYQB0AGkAbgBnAEYAbwBuAHQAIABnAGUAbgBlAHIAYQB0AGUAZAAgAGIAeQAgAEkAYwBvAE0AbwBvAG4ALgADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA) format("truetype"),url(data:application/font-woff;charset=utf-8;base64,d09GRk9UVE8AAASwAAoAAAAABGgAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABDRkYgAAAA9AAAAS0AAAEtFpovuE9TLzIAAAIkAAAAYAAAAGAIIweQY21hcAAAAoQAAABMAAAATA984gpnYXNwAAAC0AAAAAgAAAAIAAAAEGhlYWQAAALYAAAANgAAADb/0IhHaGhlYQAAAxAAAAAkAAAAJAKZAedobXR4AAADNAAAABgAAAAYAm4AEm1heHAAAANMAAAABgAAAAYABlAAbmFtZQAAA1QAAAE8AAABPPC1n05wb3N0AAAEkAAAACAAAAAgAAMAAAEABAQAAQEBB3JhdGluZwABAgABADr4HAL4GwP4GAQeCgAZU/+Lix4KABlT/4uLDAeLa/iU+HQFHQAAAHkPHQAAAH4RHQAAAAkdAAABJBIABwEBBw0PERQZHnJhdGluZ3JhdGluZ3UwdTF1MjB1RjBEOXVGMERBAAACAYkABAAGAQEEBwoNVp38lA78lA78lA77lA773Z33bxWLkI2Qj44I9xT3FAWOj5CNkIuQi4+JjoePiI2Gi4YIi/uUBYuGiYeHiIiHh4mGi4aLho2Ijwj7FPcUBYeOiY+LkAgO+92L5hWL95QFi5CNkI6Oj4+PjZCLkIuQiY6HCPcU+xQFj4iNhouGi4aJh4eICPsU+xQFiIeGiYaLhouHjYePiI6Jj4uQCA74lBT4lBWLDAoAAAAAAwIAAZAABQAAAUwBZgAAAEcBTAFmAAAA9QAZAIQAAAAAAAAAAAAAAAAAAAABEAAAAAAAAAAAAAAAAAAAAABAAADw2gHg/+D/4AHgACAAAAABAAAAAAAAAAAAAAAgAAAAAAACAAAAAwAAABQAAwABAAAAFAAEADgAAAAKAAgAAgACAAEAIPDa//3//wAAAAAAIPDZ//3//wAB/+MPKwADAAEAAAAAAAAAAAAAAAEAAf//AA8AAQAAAAEAADfYOJZfDzz1AAsCAAAAAADP/aPuAAAAAM/9o+4AAAAAALcBbgAAAAgAAgAAAAAAAAABAAAB4P/gAAACAAAAAAAAtwABAAAAAAAAAAAAAAAAAAAABgAAAAAAAAAAAAAAAAEAAAAAtwASALcAAAAAUAAABgAAAAAADgCuAAEAAAAAAAEADAAAAAEAAAAAAAIADgBAAAEAAAAAAAMADAAiAAEAAAAAAAQADABOAAEAAAAAAAUAFgAMAAEAAAAAAAYABgAuAAEAAAAAAAoANABaAAMAAQQJAAEADAAAAAMAAQQJAAIADgBAAAMAAQQJAAMADAAiAAMAAQQJAAQADABOAAMAAQQJAAUAFgAMAAMAAQQJAAYADAA0AAMAAQQJAAoANABaAHIAYQB0AGkAbgBnAFYAZQByAHMAaQBvAG4AIAAxAC4AMAByAGEAdABpAG4AZ3JhdGluZwByAGEAdABpAG4AZwBSAGUAZwB1AGwAYQByAHIAYQB0AGkAbgBnAEYAbwBuAHQAIABnAGUAbgBlAHIAYQB0AGUAZAAgAGIAeQAgAEkAYwBvAE0AbwBvAG4ALgADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA) format("woff");
 font-weight: 400;
font-style: normal
}

**Reply by palashkaria** [reply]
@DaljeetSingh1320 I faced same problem on the same stack (but different app). In this case, I had to remove a little more CSS. What you can/cannot remove totally depends on what you are using. I'd say try removing things & see if it has any impact on your app.

**Comment by 4pOnk** [comment]
I had this issue when I forgot to change permissions in downloaded font (using chmod in linux)  and tried to load it in CSS.

**Comment by Emircanaltay** [comment]
I am getting the same error what should I do?

**Reply by palashkaria** [reply]
I'm sure this is not the best way, but try deleting things: https://github.com/webpack/webpack/discussions/17101#discussioncomment-6728894

I came back to this after exactly one year, and the same solution worked on a different app

**Comment by alexander-akait** [comment]
Please provide reprducible test repo if you need help

**Reply by IlayBacil1999** [reply]
I encountered this error and created a minimal React app that reproduces it when running `npm start`. You can find the repository [here](https://github.com/IlayBacil1999/stacks-editor-bug/tree/main). I hope I did it correctly, as this is my first time participating in a discussion.

**Reply by alexander-akait** [reply]
@IlayBacil1999 thank you, reproduced

**Reply by alexander-akait** [reply]
There is a fix - https://github.com/webpack/webpack/pull/18904, release will be soon (today/tomorrow)
